### PR TITLE
Don't add a PDep reaction to the model if it exists as an unflagged LibraryReaction

### PR DESCRIPTION
**Pull request by alongd**
Addresses #741, #394 

This still does not address Josh's comment from #566, and in fact will not allow a second network reaction passing through a different isomer to be included in the model. Let's discuss how we'd like to handle that.

(tag @jimchu10, who expressed interest in this fix)

### CODE REVIEW

**Comment by alongd** [comment]
I changed this fix so it only checks whether a PDep reaction already exists as a LibraryReaction. This does not conflict with cases where the same reaction is generated via a different networks (which should still be addressed, see #566).

One thing to consider is that sometimes the same reaction can have both PDep and non-PDep pathways. If a library reaction represents only the non-PDep pathway, then the network reaction should be kept. This is still not the case in the present PR.
(examples of this situation are HOSO2+O2<=>SO3+HO2, and SH+SH<=>H2S+S -- where both an H-abstraction channel and a PDep channel exist.)

The above could be solved by tagging such reactions in our database.

**Comment by alongd** [comment]
@mliu49, this PR is now ready for review.
I didn't find an appropriate place for a unit test, let me know if you think of something.

**Comment by mliu49** [comment]
Sorry, I meant #741.

**Comment by alongd** [comment]
@jimchu10, Could you try running your model after cherry-picking the commits from this PR? We'd like to know how many instances of PDep reactions by RMG were _not_ added to your model due to this fix (can be traced by searching for `matched an existing core reaction` in the log file)
Thanks!

**Comment by jimchu10** [comment]
Sure. I will cherry-pick this commit and run the RMG P-dep job. The job might take more than a day to converge, and I will update the results once I get the final mechanism.

**Comment by jimchu10** [comment]
[out.txt](https://github.com/ReactionMechanismGenerator/RMG-Py/files/1844889/out.txt)

The log files is attached. The message "matched an existing core reaction" appear 8 time in the file. 

Although this job crashed due some other issues after 37 hours, I can still see the chemistry found is different from the output mechanism from the same RMG-Py and RMG-database without the commits in this issue. 

For example, benzyl didn't show up in previous fun, but it did appear in this run. The main reason should be that previous duplicate P-dep and library reactions made the rates over-estimated, and thus affect the later iterations of mechanism generation.

**Comment by alongd** [comment]
Thanks for sharing your log file, @jimchu10, I'm glad to hear this fix improved the reactivity of your system (assuming that the appearance of benzyl was desired...)! I also experience an improvement in the predictions of my sulfur systems using this PR.

I went through the reactions in the attached log file to make sure they already include the PDep route:
- `HCO+O=CO2+H` taken from FFCM-1. Originally from [Baulch 2005](https://aip.scitation.org/doi/pdf/10.1063/1.1748524), it is based on experimental data so it considers all pathways at the relevant conditions.
- `CH3+HO2=CH3O+OH` taken from FFCM-1. Rate constant expression uses the 1000K result (95%) from flow reactor experiments and modelling of Scire et al.(2001) and the temperature dependence from the theoretical study by [Zhu & Lin(2001)](https://pubs.acs.org/doi/pdf/10.1021/jp010698i), which is a PDep calculation done at G2M.
- `CH2OH+O2=CH2O+HO2` from FFCM-1. Originally from [Baulch 2005](https://aip.scitation.org/doi/pdf/10.1063/1.1748524), based on experimental data.
- `C2H5+O=CH2O+CH3` from FFCM-1. Originally from [Baulch 2005](https://aip.scitation.org/doi/pdf/10.1063/1.1748524), based on experimental data.
- `C2H5+O2=C2H4+HO2` from FFCM-1. Originally from [Baulch 2005](https://aip.scitation.org/doi/pdf/10.1063/1.1748524), based on experimental data.
- `CH2OH+O2=CH2O+HO2` from FFCM-1. Originally from [Baulch 2005](https://aip.scitation.org/doi/pdf/10.1063/1.1748524), based on experimental data.
- `CH3CO+O2=CH2CO+HO2` from FFCM-1. Estimated equal to the CH2CHO+O2 rate constant, which is taken from [Baulch 2005](https://aip.scitation.org/doi/pdf/10.1063/1.1748524), based on experimental data.
- `HCO+OH=CO+H2O` from FFCM-1. Originally from [Baulch 2005](https://aip.scitation.org/doi/pdf/10.1063/1.1748524), based on experimental data.

In conclusion, although the entries are only taken from FFCM and are not representative to all RMG's libraries, all reactions that were kept from a library and were not supplemented with an RMG network reaction do take into account the PDep route, which justifies our approach in this PR. I can also confirm that this is the case in my sulfur system.

In my case RMG captured 7 instances where a network reaction was not added (out of ~600 rxns), and here there are 8 instances out of an unfinished model of ~4300 rxns.

@mliu49, I'm rebasing this, let me know if you think additional discussion is needed.

**Comment by mliu49** [comment]
Just ran RMG-tests. Everything looked good except for eg1. The odd thing is that eg1 doesn't even have pdep on. Do you have any ideas for what could have caused the difference?

[eg1.log](https://github.com/ReactionMechanismGenerator/RMG-Py/files/1857464/eg1.log)

**Comment by jimchu10** [comment]
Thank you Alon. I totally agree with you that this PR match the approach we would like to do for P-dep RMG job. I will try more jobs which would make RMG jobs actually converge, and then compare the detailed modeling results of the mechanisms with/without this PR.
